src/web/signer.ts

```diff
@@ -157,3 +172,146 @@ export class TurboWebArweaveSigner extends TurboDataItemAbstractSigner {
     return super.signData(dataToSign);
   }
 }
+
+export async function streamSigner(
+  {
+    input,
+    signer,
+    fileSize,
+    emitter,
+  }: {
+    input: ReadableStream<Uint8Array> | Buffer;
+    signer: Signer;
+    fileSize: number;
+    emitter?: TurboEventEmitter;
+  },
+  opts?: DataItemCreateOptions,
+): Promise<ReadableStream<Uint8Array>> {
+  const header = createData('', signer, opts);
+
+  const stream =
+    input instanceof ReadableStream
+      ? input
+      : new ReadableStream({
+          start(controller) {
+            controller.enqueue(input);
+          },
+        });
+
+  const [s1, s2] = stream.tee(); // Clone the input stream
```

Comment:
for `node` - we setup a Readable that has the signing events on it before it is passed to `streamSign`. here you are moving that down to `deepHash` and not providing end/failure events. 

instead - i think we should provide a custom ReadableStream that has the events on it, so as the stream signer pulls bytes through, those events are emitted. an example of this is in wayfinder, where we emit verification events on the stream returned to the client - https://github.com/ar-io/ar-io-sdk/blob/beta/src/common/wayfinder/wayfinder.ts#L289-L365

Reply:
https://github.com/ardriveapp/turbo-sdk/pull/259/files#r2114908658